src/JsonSchema/SchemaFactory.php

```diff
@@ -230,14 +233,33 @@ private function buildPropertySchema(Schema $schema, string $definitionName, str
                 break;
             }
 
-            if ($type->isNullable()) {
-                $propertySchema['anyOf'] = [['$ref' => $subSchema['$ref']], ['type' => 'null']];
-            } else {
-                $propertySchema['$ref'] = $subSchema['$ref'];
+            $isNullable = $isNullable ?? $type->isNullable();
+            $refs[$subSchema['$ref']] = '$ref';
```

Comment:
Value is never used as you're always calling `array_keys($refs)`. I guess you did it to ensure unicity, so I suggest the following:
```suggestion
            $ref = $subSchema['$ref'];
            if (!\in_array($ref, $refs, true)) {
                $refs[] = $ref;
            }
```
Then you can just use `$refs` instead of `array_keys($refs)`

Reply:
You can also update `$isNullable` as following:
```php
if ($type->isNullable() && !\in_array(['type' => 'null'], $refs, true)) {
    $refs[] = ['type' => 'null'];
}
```

Reply:
With those updates, you won't need `$isNullable` anymore on line 247, and won't need `$anyOf` neither

Reply:
Pushed the changes. Thank you so much for thinking this though with me, it is already a much better outcome!